File: src/jvmMain/java/core/Ignore.java
package core;

/* Simone 04/09/13 11.30 */

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Ignore {
    final static Logger log = LoggerFactory.getLogger(Ignore.class);

    public static void exception(RunnableEx runnable) {
        try {
            runnable.run();
        } catch (Exception ex) {
            log.error("", ex);
        }
    }

    public static void exception(Runnable runnable) {
        try {
            runnable.run();
        } catch (Exception ex) {
            log.error("", ex);
        }
    }
}

File: src/jvmMain/kotlin/core/Ignore.kt
package core

import org.slf4j.LoggerFactory

/* Simone 04/09/13 11.30 */   object Ignore {
    val log = LoggerFactory.getLogger(Ignore::class.java)
    fun exception(runnable: RunnableEx) {
        try {
            runnable.run()
        } catch (ex: Exception) {
            log.error("", ex)
        }
    }

    @JvmStatic
    fun exception(runnable: Runnable) {
        try {
            runnable.run()
        } catch (ex: Exception) {
            log.error("", ex)
        }
    }

    @JvmStatic
    fun exception(runnable: () -> Unit) {
        try {
            runnable()
        } catch (ex: Exception) {
            log.error("", ex)
        }
    }
}